packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +324,74 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	it("should not call entry point before data store is requested", async () => {
+		const dataStoreFactory1 = new DataObjectFactory(
+			TestDataObjectType1,
+			TestDataObject1,
+			[],
+			[],
+		);
+		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+			runtime.IFluidHandleContext.resolveHandle(request);
+		const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+			[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+		]);
+		const containerRuntimeFactoryWithDefaultDataStore =
+			getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+		const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[innerRequestHandler],
+		);
+		const loader = provider.createLoader([[provider.defaultCodeDetails, runtimeFactory]]);
+
+		// Create a container for the first client.
+		const container1 = await createAndAttachContainer(
+			provider.defaultCodeDetails,
+			loader,
+			provider.driver.createCreateNewRequest(provider.documentId),
+		);
+		assert.doesNotThrow(
+			async () => requestFluidObject<TestDataObject1>(container1, "/"),
+			"Initialization from container client should success.",
+		);
+
+		// Create a summarizer for the container and do a summary shouldn't throw.
+		const createSummarizerResult = await createSummarizerFromFactory(
+			provider,
+			container1,
+			dataStoreFactory1,
+			undefined,
+			containerRuntimeFactoryWithDefaultDataStore,
+			registryStoreEntries,
+		);
+		const summarizer = createSummarizerResult.summarizer;
+		assert.doesNotThrow(() => {
+			summarizer.summarizeOnDemand({ reason: "test" });
+		});
```

Comment:
Oh, one more thing here. @agarwal-navin in order to validate that the summary really did complete without issues, we should do [all this](https://github.com/microsoft/FluidFramework/blob/main/packages/test/test-end-to-end-tests/src/test/summaries.spec.ts#L132-L149) (skipping the `negResult`), correct? Awaiting all the promises in the object returned by `summarizeOnDemand()`.

Reply:
Yes, you need to validate that the summary was successful. Also, the result needs to be awaited. You can call this [helper function](https://github.com/agarwal-navin/FluidFramework/blob/6af5a9369ca7f0a52e82e53a52d99246b649a11f/packages/test/test-utils/src/TestSummaryUtils.ts#L134) which will do all the validations.